flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepointTest.java

```diff
@@ -82,7 +82,9 @@ void testJobFailed() throws Exception {
         try (MockStopWithSavepointContext ctx = new MockStopWithSavepointContext()) {
             StateTrackingMockExecutionGraph mockExecutionGraph =
                     new StateTrackingMockExecutionGraph();
-            StopWithSavepoint sws = createStopWithSavepoint(ctx, mockExecutionGraph);
+            final CompletableFuture<String> savepointFuture = CompletableFuture.completedFuture("");
```

Comment:
I guess, we should extend the test name here to something like `testJobFailedWithSavepointCreationSuccessful` to distinguish the test in a better way from the other test cases.

Additionally, we should verify that a `StopWithSavepointStopException` is part of the cause for this specific test case, shouldn't we?

Reply:
> needs another explicit test for job fails + successful savepoint

is that the test you're refering to in your [comment above](https://github.com/apache/flink/pull/19968#pullrequestreview-1010663103)?

`testJobFailed` pretty much covers the scenario, doesn't it?

Reply:
> Additionally, we should verify that a StopWithSavepointStopException is part of the cause for this specific test case, shouldn't we?

> is that the test you're refering to in your https://github.com/apache/flink/pull/19968#pullrequestreview-1010663103?

yes that's what I meant. The enrichment isn't explicitly covered afaict.

Reply:
> testJobFailed pretty much covers the scenario, doesn't it?

It's close to covering it; just needs a 1 line adjustment :)